shared/src/main/scala/ast/Expression.scala

```diff
@@ -266,47 +265,47 @@ case class LogicalNegation(children: List[Expression]) extends Expression("lneg"
   def withChildren(newChildren: List[Expression]): Expression = copy(children = newChildren)
 }
 
-case class FocalMax(children: List[Expression], neighborhood: Neighborhood, target: TargetCell) extends Expression("fmax") with FocalExpression {
+case class FocalMax(children: List[Expression], neighborhood: Neighborhood, target: Option[TargetCell] = Some(TargetCell.All)) extends Expression("fmax") with FocalExpression {
```

Comment:
This is a bit ugly, however, it is impossible for circe to mark it as optional otherwise.

Reply:
Ah, so you can't do `target: TargetCell = TargetCell.All` without custom codecs. That's frustrating. And switching to custom encoders probably goes against convention enough to make it look "weird".

I don't have a strong sense of which is better. If you do keep this one, it could be worth aliasing TargetCell.All to some other val for all the focal operations so that each operation can call on a consistent default.

Reply:
@CloudNiner I remembered one more thing to try! fingers crossed // will probably ping you with some extra code to compare which one you like more